src/main/kotlin/no/nav/dagpenger/regel/api/KafkaDagpengerBehovConsumer.kt

```diff
@@ -0,0 +1,123 @@
+package no.nav.dagpenger.regel.api
+
+import mu.KotlinLogging
+import no.nav.dagpenger.regel.api.minsteinntekt.MinsteinntektBeregning
+import no.nav.dagpenger.regel.api.minsteinntekt.MinsteinntektBeregninger
+import no.nav.dagpenger.regel.api.minsteinntekt.MinsteinntektResultat
+import no.nav.dagpenger.regel.api.minsteinntekt.MinsteinntektResultatParametere
+import no.nav.dagpenger.regel.api.tasks.Tasks
+import no.nav.dagpenger.streams.KafkaCredential
+import no.nav.dagpenger.streams.Topics
+import no.nav.dagpenger.streams.streamConfig
+import org.apache.kafka.common.serialization.Serdes
+import org.apache.kafka.streams.KafkaStreams
+import org.apache.kafka.streams.StreamsBuilder
+import org.apache.kafka.streams.Topology
+import org.apache.kafka.streams.kstream.Consumed
+import java.time.LocalDateTime
+import java.util.Properties
+import java.util.concurrent.TimeUnit
+
+private val LOGGER = KotlinLogging.logger {}
+
+class KafkaDagpengerBehovConsumer(
+    val env: Environment,
+    val tasks: Tasks,
+    val minsteinntektBeregninger: MinsteinntektBeregninger
+) {
+
+    val SERVICE_APP_ID = "dp-regel-api"
+    private lateinit var streams: KafkaStreams
+    fun start() {
+        streams = KafkaStreams(buildTopology(), this.getConfig())
+        streams.setUncaughtExceptionHandler { t, e -> System.exit(0) }
+        streams.start()
+    }
+
+    fun stop() {
+        LOGGER.info { "Shutting down $SERVICE_APP_ID" }
+        streams.close(3, TimeUnit.SECONDS)
+        streams.cleanUp()
+    }
+
+    internal fun buildTopology(): Topology {
+        val builder = StreamsBuilder()
+
+        val stream = builder.stream(
+            Topics.DAGPENGER_BEHOV_EVENT.name,
+            Consumed.with(Serdes.StringSerde(), Serdes.serdeFrom(JsonSerializer(), JsonDeserializer()))
+        )
+
+        stream
+            .peek { key, value -> LOGGER.info("Processing behov with id ${value.behovId}") }
+            //.filter { _, behov -> shouldBeProcessed(behov) }
```

Comment:
```suggestion
            .filter { _, behov -> hasNeededMinsteinntektResult(behov) }
```
, and then remove the check in `storeResult()`?

Reply:
+1